front/components/agent_builder/AgentBuilder.tsx

```diff
@@ -10,82 +10,127 @@ import { AgentBuilderLayout } from "@app/components/agent_builder/AgentBuilderLa
 import { AgentBuilderLeftPanel } from "@app/components/agent_builder/AgentBuilderLeftPanel";
 import { AgentBuilderRightPanel } from "@app/components/agent_builder/AgentBuilderRightPanel";
 import { submitAgentBuilderForm } from "@app/components/agent_builder/submitAgentBuilderForm";
+import {
+  getDefaultAgentFormData,
+  transformAgentConfigurationToFormData,
+} from "@app/components/agent_builder/transformAgentConfiguration";
 import { useMCPServerViewsContext } from "@app/components/assistant_builder/contexts/MCPServerViewsContext";
 import { appLayoutBack } from "@app/components/sparkle/AppContentLayout";
 import { FormProvider } from "@app/components/sparkle/FormProvider";
 import { useSendNotification } from "@app/hooks/useNotification";
+import { useAgentConfigurationActions } from "@app/lib/swr/actions";
+import { useEditors } from "@app/lib/swr/editors";
 import { useFeatureFlags } from "@app/lib/swr/workspaces";
 import logger from "@app/logger/logger";
-import type { AgentConfigurationType } from "@app/types";
+import type { LightAgentConfigurationType } from "@app/types";
 import {
   EXTENDED_MAX_STEPS_USE_PER_RUN_LIMIT,
-  GPT_4O_MODEL_CONFIG,
   MAX_STEPS_USE_PER_RUN_LIMIT,
 } from "@app/types";
 
 interface AgentBuilderProps {
-  agentConfiguration?: AgentConfigurationType;
+  agentConfiguration?: LightAgentConfigurationType;
 }
 
 export default function AgentBuilder({
   agentConfiguration,
 }: AgentBuilderProps) {
   const { owner, user, supportedDataSourceViews } = useAgentBuilderContext();
-  const { mcpServerViews } = useMCPServerViewsContext();
+  const { mcpServerViews, isMCPServerViewsLoading } =
+    useMCPServerViewsContext();
   const router = useRouter();
   const sendNotification = useSendNotification();
 
+  const { actions, isActionsLoading } = useAgentConfigurationActions(
+    owner.sId,
+    agentConfiguration?.sId ?? null,
+    mcpServerViews
+  );
+
+  const { editors, isEditorsLoading } = useEditors({
+    owner,
+    agentConfigurationId: agentConfiguration?.sId ?? null,
+  });
+
   const { hasFeature } = useFeatureFlags({
     workspaceId: owner.sId,
   });
 
-  const hasExtendedFeature = hasFeature("extended_max_steps_per_run");
-  const defaultMaxSteps = hasExtendedFeature
-    ? Math.min(10, EXTENDED_MAX_STEPS_USE_PER_RUN_LIMIT)
-    : Math.min(10, MAX_STEPS_USE_PER_RUN_LIMIT);
+  const defaultMaxSteps = hasFeature("extended_max_steps_per_run")
+    ? EXTENDED_MAX_STEPS_USE_PER_RUN_LIMIT
+    : MAX_STEPS_USE_PER_RUN_LIMIT;
```

Comment:
ooc: why here we're using `MAX_STEPS_USE_PER_RUN_LIMIT` instead of the default in the `getDefaultAgentFormData` where it's `DEFAULT_MAX_STEPS_USE_PER_RUN`?